MOBT-3-Patterns/MVVMWithoutC/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  LoginViewModel.swift
+//  MVVMWithoutC
+//
+//  Created by Mykyta Babanin on 01.07.2021.
+//
+
+import Foundation
+
+
+protocol LoginViewModelProtocol {
+    typealias authenticationLoginCallBack = (_ status: Bool) -> Void
+    var modelUserName: String { get }
+    var modelPassword: String { get }
+    func authenticationUserWith(userName: String, andPassword password: String)
+    func verifyUserWith(_ userName: String, andPassword password: String)
+    func loginCompletion(callBack: @escaping authenticationLoginCallBack)
+    func displaySignUpPage()
+    func displaySignOut()
+    func viewDidDisapear()
+}
+
+class LoginViewModel: LoginViewModelProtocol {
+    weak var coordinator: LoginCoordinator?
+    let defaults: UserDefaults
+    let model: LoginModel
+    
+    var modelUserName: String {
+        get {
+            guard let userName = model.userName else { return "" }
```

Comment:
You can simply use the [nil coalescing operator](https://www.hackingwithswift.com/example-code/language/what-is-the-nil-coalescing-operator):
```suggestion
            return model.userName ?? ""
```